sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -33,6 +33,73 @@ import org.apache.spark.sql.catalyst.util.{ArrayBasedMapData, GenericArrayData}
 import org.apache.spark.sql.types._
 
 /**
+ * Common base class for [[StaticInvoke]], [[Invoke]], and [[NewInstance]].
+ */
+trait InvokeLike extends Expression {
+
+  def arguments: Seq[Expression]
+
+  def propagateNull: Boolean
+
+  def prepareArguments(ctx: CodegenContext, ev: ExprCode): (String, String, String) = {
+
+    val argsHaveNull =
+      if (propagateNull && arguments.exists(_.nullable)) {
```

Comment:
`propagateNull && arguments.exists(_.nullable)` is used many times, define a variable for it?